<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>611</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>270</y>
      <width>491</width>
      <height>321</height>
     </rect>
    </property>
    <property name="title">
     <string>Dialog</string>
    </property>
    <widget class="QTextBrowser" name="textBrowser">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>471</width>
       <height>281</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="Settings">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>431</width>
      <height>251</height>
     </rect>
    </property>
    <property name="title">
     <string>Settings</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>391</width>
       <height>201</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Hash value to crack</string>
        </property>
        <widget class="QLineEdit" name="HashValue">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>30</y>
           <width>341</width>
           <height>41</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="SelectionSetting">
        <property name="title">
         <string>Generation/Crack Settings</string>
        </property>
        <widget class="QWidget" name="horizontalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>19</y>
           <width>391</width>
           <height>81</height>
          </rect>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QGroupBox" name="AlgorithmGroup">
            <property name="title">
             <string>Algorithm</string>
            </property>
            <widget class="QRadioButton" name="MD5Sel">
             <property name="geometry">
              <rect>
               <x>20</x>
               <y>20</y>
               <width>115</width>
               <height>19</height>
              </rect>
             </property>
             <property name="text">
              <string>MD5</string>
             </property>
            </widget>
            <widget class="QRadioButton" name="SHA1Sel">
             <property name="geometry">
              <rect>
               <x>20</x>
               <y>50</y>
               <width>115</width>
               <height>19</height>
              </rect>
             </property>
             <property name="text">
              <string>SHA1</string>
             </property>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="ContainerGroup">
            <property name="title">
             <string>Container</string>
            </property>
            <widget class="QRadioButton" name="MapSel">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>20</y>
               <width>115</width>
               <height>19</height>
              </rect>
             </property>
             <property name="statusTip">
              <string>Choose std::map as storage method</string>
             </property>
             <property name="text">
              <string>Map</string>
             </property>
            </widget>
            <widget class="QRadioButton" name="HashSel">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>50</y>
               <width>115</width>
               <height>19</height>
              </rect>
             </property>
             <property name="statusTip">
              <string>Choose Hash as stroage method</string>
             </property>
             <property name="text">
              <string>Hash</string>
             </property>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="Range">
            <property name="title">
             <string>Range</string>
            </property>
            <widget class="QSpinBox" name="lowMargin">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>30</y>
               <width>46</width>
               <height>22</height>
              </rect>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
            </widget>
            <widget class="QSpinBox" name="upMargin">
             <property name="geometry">
              <rect>
               <x>80</x>
               <y>30</y>
               <width>46</width>
               <height>22</height>
              </rect>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>10000</number>
             </property>
             <property name="value">
              <number>20</number>
             </property>
            </widget>
            <widget class="QLabel" name="connectionLabel">
             <property name="geometry">
              <rect>
               <x>60</x>
               <y>30</y>
               <width>16</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>-</string>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>491</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>One Step Crack Demonstration</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>232</width>
       <height>191</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="FrontNode1">
        <property name="text">
         <string>Front Node 1</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="ConstLabel2">
        <property name="text">
         <string>............</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="FrontNode2">
        <property name="text">
         <string>Front Node 2 </string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="EndNode2">
        <property name="text">
         <string>End Node 2</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="EndNode1">
        <property name="text">
         <string>End Node 1</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="ConstLabel1">
        <property name="text">
         <string>............</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="FrontNodeEnd">
        <property name="text">
         <string>Front Node End</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="EndNodeEnd">
        <property name="text">
         <string>End Node End</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="ConstHash1">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>40</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>GivenHash</string>
     </property>
    </widget>
    <widget class="QLabel" name="ConstR31">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>20</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>R3</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="Value1">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>40</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>Value</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="OKOrNot1">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>20</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>   OK?</string>
     </property>
    </widget>
    <widget class="QLabel" name="ConstHash2">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>130</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>     Hash</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>60</y>
       <width>111</width>
       <height>42</height>
      </rect>
     </property>
     <property name="text">
      <string>............</string>
     </property>
    </widget>
    <widget class="QLabel" name="ConstR32">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>100</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>R3</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="Rx">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>100</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>R...</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="OKOrNot2">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>100</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>OK?</string>
     </property>
    </widget>
    <widget class="QLabel" name="ConstFrontNode">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>180</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Front Node</string>
     </property>
    </widget>
    <widget class="QLabel" name="ConstHash3">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>180</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>Hash</string>
     </property>
    </widget>
    <widget class="QLabel" name="StopOrNot">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>210</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>    </string>
     </property>
    </widget>
    <widget class="QLabel" name="RealValue">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>180</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>RealValue</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>270</y>
      <width>431</width>
      <height>311</height>
     </rect>
    </property>
    <property name="title">
     <string>Control Unit</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>-20</x>
       <y>20</y>
       <width>471</width>
       <height>281</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0">
       <widget class="QPushButton" name="OSCD">
        <property name="minimumSize">
         <size>
          <width>233</width>
          <height>60</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Embark on a demonstration step-by-step</string>
        </property>
        <property name="text">
         <string>One Step Crack Demonstration</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="Statistics">
        <property name="minimumSize">
         <size>
          <width>144</width>
          <height>60</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Show how long the last crack  took</string>
        </property>
        <property name="text">
         <string>Statistics</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="StartCrack">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>168</horstretch>
          <verstretch>60</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>144</width>
          <height>60</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>144</width>
          <height>60</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Crack the given hash value with currently loaded rainbow table</string>
        </property>
        <property name="text">
         <string>Start Crack</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="StartGeneration">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>88</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>233</width>
          <height>60</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>233</width>
          <height>60</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Start generating a rainbow table with the current settings</string>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>Start Generation</string>
        </property>
        <property name="shortcut">
         <string>Return</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
